File: frontend/src/domain/entities/Pago_entity.ts
import { TimeStamp } from "../values/Timestamp";

export class PagoEntity extends TimeStamp {
    id: number | null;
    monto: number;
    moneda: string;
    estado: string;
    carrito: number;
    payment_method_id: string;
    payment_intent: string;

    constructor(id: number | null, monto: number, moneda: string, estado: string, carrito: number, payment_method_id: string, payment_intent: string,
        created_at: Date | null = null,
        updated_at: Date | null = null,
        deleted_at: Date | null = null
    ) {
        super(created_at, updated_at, deleted_at);
        this.id = id;
        this.monto = monto;
        this.moneda = moneda;
        this.estado = estado;
        this.carrito = carrito;
        this.payment_method_id = payment_method_id;
        this.payment_intent = payment_intent;
    }
}

File: backend/app_api/domain/entities/Pago_entity.py
from dataclasses import dataclass
from app_api.domain.values.timestamp import TimestampModel

@dataclass(kw_only=True)
class PagoEntity(TimestampModel):
    id: int | None
    monto: int
    moneda: str
    estado: str
    carrito: int
    payment_method_id: str
    payment_intent: str
